doc/howtos/bluetoothspeaker.rst

```diff
@@ -0,0 +1,136 @@
+
+  [Install]
+  WantedBy=multi-user.target
+
+Copy the pulseaudio.service file to::
+
+  /etc/systemd/system/
+
+and enable it by::
+
+  systemctl enable pulseaudio.service
+
+For this particular use case we are starting pulseaudio in system mode
+and including bluetooth modules to the system.pa file. Remember that
+starting pulseaudio in system mode is not the recommended way and you
+should be aware of the possible consequences.
+
+system.pa can be found from::
+
+  /etc/pulse/
+
+Add the following lines to the system.pa to enable bluetooth::
+
+  .ifexists module-bluetooth-policy.so
+  load-module module-bluetooth-policy
+  .endif
+
+  .ifexists module-bluetooth-discover.so
+  load-module module-bluetooth-discover
+  .endif
+
+Add D-Bus configuration so that pulseaudio can communicate with BlueZ
+=====================================================================
+
+Copy D-Bus configuration file (name it for example as
+pulseaudio-bluetooth.conf) to::
+
+  /etc/dbus-1/system.d/
+
+ Contents of the file should be the following::
+
+   <busconfig>
+     <policy user="pulse">
+       <allow send_destination="org.bluez"/>
+     </policy>
+   </busconfig>
+
+Configure Bluez to enable A2DP sink
+===================================
+
+Bluez needs configuration to enable A2DP sink feature.
+This can be done by copying file audio.conf to::
+
+  /etc/bluetooth/
+
+With following contents::
+
+  [General]
+  Enable=Source
+
+Unblock bluetooth at boot with Connman
+======================================
+
+By default the bluetooth radio is soft blocked. To enable it in the
+boot you can change Connman settings by changing the folllowing line::
+
+  [Bluetooth]
+  Enable=false
+
+to::
+  
+  [Bluetooth]
+  Enable=true
+
+Pair and connect your source device
+===================================
+
+For pairing you can use Bluez tools of your flavor. For example with
+bluetoothctl you can set the bluetooth chip to be discoverable and
+scan the surrounding devices. Bluez test tools also includes
+simple-agent (Python) which you can use for easy pairing. Starting the
+simple-agent with --noinputoutput option enables you to pair and
+connect without user input from the bluetooth speaker. When you are
+paired and connected pulseaudio will get notification from D-Bus and
+setup the a2dp audio path automatically and you can just start
+playback from your connected device.
+
+If you want to build your own speaker pairing software one good choice
+is to use Python with D-Bus (to communicate to BlueZ) and mraa to get
+button presses for simplified UI experience. Check BlueZ simple-agent
+and mraa Python examples for ideas how to do this.
+
+If you plan to use above Python tools you need to add the following
+to your local.conf::
+
+  REFKIT_IMAGE_GATEWAY_EXTRA_INSTALL_append = " \
+    python-dbus \
+    python-pygobject \
+    bluez5-testtools"
+
+And also to meta-refkit/conf/distro/include/refkit-supported-recipes.txt::
```

Comment:
At this point I don't know. There is also problem with yocto packaging that it is very difficult to update other packages configuration files with different apps recipe (like pulseaudio, connman and bluez). It could be done with a separate layer (I quess), but it seems like an overkill. So there is no very clean way atm to offer this configuration to the users so that it doesn't break future updates and you could actually nicely remove it. As mentioned before, I have a experimental setup with linux user, I could try that with a recipe (or maybe even Krisztians flatpack setup). However that problem is kind of general in nature and I can't solve it with this bt thing.